Global Search - Non taxon search uses ALA general search

tony-ross opened the pull request:
Fixes autocomplete functionality and improve BieService query handling
- Updated autocomplete configuration to include navbar search input.
- Integrated autocomplete configuration into nbn.js.
- Refactored BieService to improve general query content handling

tony-ross committed:
Enhance autocomplete functionality and improve BieService query handling

- Updated autocomplete configuration to include navbar search input.
- Integrated autocomplete configuration into nbn.js.
- Refactored BieService to improve general query context handling

tony-ross commented:
You can get round the hard coded url in the header template and do a global search by issuing the non-taxon search directly in the browser url: 
---
![image](https://github.com/user-attachments/assets/4959b896-5722-436d-84dd-48b6384663ba)

h-m-jones commented on grails-app/services/uk/org/nbn/bie/BieService.groovy:
```diff
@@ -107,115 +107,126 @@ class BieService extends au.org.ala.bie.BieService{
-                if (occFilter) {
-                    queryContext = queryContext + "%20AND%20" + URIUtil.encodeWithinQuery(occFilter).replaceAll("%26","&").replaceAll("%3D","=").replaceAll("%3A",":")
+        if (requestObj.fq == "idxtype:TAXON") {
+            log.info("Performing NBN taxon search")
```
change to log.debug  (and the one below)

tony-ross committed:
Update version number to 1.5.2-NBN-0-0-10-SNAPSHOT in build.gradle

tony-ross commented:
On the autocomplete I was getting this error message "Error from server at http://localhost:8985/solr/bie: suggester was not built" 

Here's how I fixed it:

` curl http://localhost:8985/solr/bie/suggest\?suggest.build\=true

{
  "responseHeader":{
    "status":0,
    "QTime":16306},
  "command":"build"}`
}

tony-ross committed:
Fixed autocomplete not working
- Added global search configuration variable to GSP layout SEARCH_CONF
- Updated layout to include bieWebServiceUrl for enhanced search functionality.

tony-ross committed:
Change log level for NBN taxon search in BieService from INFO to DEBUG

h-m-jones commented on grails-app/services/uk/org/nbn/bie/BieService.groovy:
```diff
@@ -107,115 +107,126 @@ class BieService extends au.org.ala.bie.BieService{
+            }
+
+        } else {
+            log.info("Performing general search across all fields")
```
change to debug too (otherwise the log file will fill up)

tony-ross committed:
use DEBUG level for query URLs and search operations

tony-ross committed:
changed log levels

tony-ross committed:
Reverted  autocomplete configuration

tony-ross commented:
@h-m-jones autocomplete reverted from this PR ( now done in layout ).

tony-ross committed:
Update BieService.groovy

tony-ross committed:
Update BieService.groovy

tony-ross committed:
Update autocomplete-configuration.js

tony-ross committed:
Update autocomplete-configuration.js

tony-ross committed:
Update autocomplete-configuration.js

h-m-jones committed:
removed whitespace changes
